### ISSUE TITLE
hash txt error

### ISSUE THREAD

**Issue by antonzhelyazkov** [ftpgrab/ftpgrab#34]
Hi,

ftpgrab is nice software. I use it with hash storage text. I have got error

```--------------
/usr/local/ftpgrab/ftpgrab.sh: line 73: [: 76cb7ce2867a671c86bd35f0697a902b: binary operator expected
Process file: 24techno/EPG-12-01-2018.xml
Hash: 76cb7ce2867a671c86bd35f0697a902b
Size: 5.34 Kb
Status: Already downloaded and valid. Skip download...
Time spent: 00:00:01
 ```

**Comment by crazy-max** [comment]
Hi @antonzhelyazkov, please review your issue : https://ftpgrab.github.io/doc/reporting-issue/
Thanks

**Comment by antonzhelyazkov** [comment]
Hi,

I tried to investigate the problem. We have strange circumstances - files in different directories have same names! For example:

Process file: animalplanet/EPG-14-01-2018.xml
Process file: discovery/EPG-14-01-2018.xml
Process file: natgeo/EPG-14-01-2018.xml
Process file: eurosport/EPG-14-01-2018.xml
Process file: 24kitchen/EPG-14-01-2018.xml

Problem occurs when files are already downloaded

Here are other information about our system

```
awk -W version
GNU Awk 3.1.7
```

configuration file

```
cat /usr/local/ftpgrab/conf/all.conf

DIR_DEST="/path/to/dir/"
EMAIL_LOG="my@email.com"
DEBUG=0

FTP_HOST="xxx.xxx.xxx.xxx"
FTP_PORT="21"
FTP_USER="1qaz"
FTP_PASSWORD="qwerty"
FTP_SOURCES="/"

FTP_SECURE=0
FTP_CHECK_CERT=0

DL_METHOD="wget"
DL_USER="epg"
DL_GROUP="epg"
DL_CHMOD=""
DL_REGEX=""
DL_EXCLUDE_REGEX=""
DL_RETRY=3
DL_RESUME=0
DL_SHUFFLE=0
DL_HIDE_SKIPPED=0
DL_HIDE_PROGRESS=1
DL_CREATE_BASEDIR=0

HASH_ENABLED=1
HASH_TYPE="md5"
HASH_STORAGE="text"
```

#### Configuration

**Operating system and platform** : Centos 6 64bits

**ftpgrab version** : 4.3

**wget version** : 1.12

**curl version** : 7.19.7

**md5sum / sha1sum version** : 8.4

**sqlite3 version** : 3.6.20

**Comment by crazy-max** [comment]
Hi, can you reproduce the problem with DEBUG=1 in the config fileand post your log file please ?
Thanks

**Comment by antonzhelyazkov** [comment]
Hi @crazy-max,

here is output with debug mode

```
--------------
#DEBUG lineClean: /24techno/
#DEBUG basename: 24techno
#DEBUG srcfile: /24techno
#DEBUG srcfileproc: /24techno
#DEBUG srcfileshort: 24techno
#DEBUG srcfileshort2: 24techno
#DEBUG srchash: 564aa1020059848f23e017d436a42f4a
#DEBUG srcsize:
#DEBUG destfile: /home/epg/ftp/24techno
#DEBUG destsize: N/A
#DEBUG vregex: 24techno
#DEBUG lineClean: /24techno/EPG-05-01-2018.xml
#DEBUG basename: EPG-05-01-2018.xml
#DEBUG srcfile: /24techno/EPG-05-01-2018.xml
#DEBUG srcfileproc: /24techno/EPG-05-01-2018.xml
#DEBUG srcfileshort: 24techno/EPG-05-01-2018.xml
#DEBUG srcfileshort2: 24techno/EPG-05-01-2018.xml
#DEBUG srchash: e2a6a52362d400241cbb94fcf242e071
#DEBUG srcsize: 5748
#DEBUG destfile: /home/epg/ftp/24techno/EPG-05-01-2018.xml
#DEBUG destsize: 5748
#DEBUG vregex: 24techno/EPG-05-01-2018.xml
/usr/local/ftpgrab/ftpgrab.sh: line 73: [: e2a6a52362d400241cbb94fcf242e071: binary operator expected
Process file: 24techno/EPG-05-01-2018.xml
Hash: e2a6a52362d400241cbb94fcf242e071
Size: 5.61 Kb
Status: Already downloaded and valid. Skip download...
Time spent: 00:00:00
```

**Comment by crazy-max** [comment]
Ok thx i keep u in touch

**Comment by crazy-max** [comment]
Can you post your hash file please ? It seems there is unexpected symbols.

**Comment by crazy-max** [comment]
Nevermind i found the issue

**Comment by antonzhelyazkov** [comment]
Do you want to setup FTP with our or send you our ftp root with all files to reproduce the problem at your site?

**Comment by crazy-max** [comment]
No thanks @antonzhelyazkov, i've got everything i need.
Will be solved in the next release.

**Comment by crazy-max** [comment]
Has been solved in the latest release (4.3.1)

**Comment by antonzhelyazkov** [comment]
Great :)
Now script works correctly.
